lib/SymbolGraphGen/JSON.cpp

```diff
@@ -72,6 +72,27 @@ void swift::symbolgraphgen::serialize(const ExtensionDecl *Extension,
       if (const auto *ExtendedModule = ExtendedNominal->getModuleContext()) {
         OS.attribute("extendedModule", ExtendedModule->getNameStr());
       }
+
+      StringRef Kind;
+      switch (ExtendedNominal->getKind()) {
```

Comment:
Is there no existing API for doing this? as in converting a `DeclKind` to a string suitable for symbol graph output? If not this would be a good opportunity to refactor this code.

Reply:
Please include the `swift.` prefix. I will at some point add similar functionality to clang for Objective-C categories, when we do that it might make sense to unify the SGF representation for both concepts since they are extremely similar.
WRT refactoring this with `Symbol.getKind(Decl *)`, I had a look at that function and it feels like it should be a `static` in Symbol as opposed to a member function. If you do that it might make sense to just make this chunk of code something like:
```
switch (ExtendedNominal->getKind()) {
    case DeclKind::Class:
    case DeclKind::Struct:
    case DeclKind::Enum:
    case DeclKind::Protocol:
        Kind = Symbol::getKind(ExtendedNominal);
        break;
    default:
        Kind = "unknown"
        break;
}
```

Reply:
Resolved in 9af13ab80bdb660e89f70b6c9dd64171bcea378e. I didn't add the extra switch filter in JSON.cpp as I think the JSON serialization shouldn't have authority over what kinds are valid in the context of `swiftExtension.typeKind`.